### ISSUE TITLE
Unable to checkout

### ISSUE THREAD

**Issue by OgreMHDW** [magento/magento2#4333]
## Steps to reproduce
1. Install Magento.
2. Add Product to the cart.
3. Select Checkout
## Expected result
1. Magento should take to https checkout page.
## Actual result
1. Takes the user to the http account page.
2. No products are stored in the shopping cart.
3. Redirecting back to the http OR https version of the site shows the shopping cart but does not allow checkout.

URL: http://www.capricornpacificlv.com

**Comment by pboisvert** [comment]
Did you make sure you have Allow Guest Checkout set to "Yes" under Store > Configuration > Sales > Checkout > Checkout Options?

**Comment by ishakhsuvarov** [comment]
@OgreMHDW Thank you for reporting. Please let us know how did you configure checkout.

**Comment by OgreMHDW** [comment]
"Allow Guest Checkout" is not enabled. "Require Customer To Be Logged In To Checkout" is enabled.

**Comment by OgreMHDW** [comment]
So I think I may have figured something out. This appears to be an issue with the "Enable Onepage Checkout" option. I have set onepage checkout to No and now am able to view and edit cart without it being emptied. Unfortunately, I have no way to check out at that point.

**Comment by OgreMHDW** [comment]
Here is the view cart setting with Enable Onepage Check enabled:

![2016-04-29_084724](https://cloud.githubusercontent.com/assets/4369476/14921529/8795caea-0de7-11e6-9068-2c3a888a872a.png)

Once I select Go To Checkout, it takes me to the account page with cart cleared.

Here is it disabled:

![2016-04-29_085041](https://cloud.githubusercontent.com/assets/4369476/14921530/8798e054-0de7-11e6-8f15-18222b5102a8.png)

I have no way of checking out at this point.

**Comment by OgreMHDW** [comment]
So an update. I figured this may have happened due to installing from Composer. I created a staging site and used the downloaded Magento 2.0.5 for installation. The site was working normally until I enabled SSL. After that I am experiencing the same issues as my other installation. Since SSL is critical to protecting passwords, this is not functional without it. No caching is being used on the new staging install.
